src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java

```diff
@@ -71,6 +75,13 @@ private boolean isFreeMediationConfirmed(Claim claim) {
             && isOptedForMediation(response);
     }
 
+    private boolean hasClaimantSettledForFullDefense(Claim claim){
+        ClaimantResponse claimantResponse = claim.getClaimantResponse().orElseThrow(IllegalStateException::new);
+        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+        return claimantResponse.getType() == ClaimantResponseType.ACCEPTATION
+            && (isResponseFullDefenceStatesPaid(response) || isFullDefenceDispute(response));
```

Comment:
We don't support counterclaiming yet - what do you think about just checking that it's a `FULL_DEFENCE` here instead of the sub-types?

Reply:
but when you do? Also, I do need the other types elsewhere

Reply:
Then the implementation doesn't reflect the method name; when counterclaiming is implemented, someone could use this method expecting it to return `true` because of the method name (the claimant has settled, and it is a full defence) when actually it returns `false` because the implementation isn't _really_ doing that